hack/run_test

```diff
@@ -335,85 +344,136 @@ function test_value_replication() {
   postgresql_cmd -c "CREATE TABLE $table_name (a integer); INSERT INTO $table_name VALUES ($value);"
 
   # Read value from slaves and check whether it is expected
-  for slave in $slave_cids; do
-    slave_ip=$(get_ip_from_cid $slave)
+  for slave_ip in $slave_ips; do
     CONTAINER_IP=$slave_ip
     for i in $(seq $max_attempts); do
       result="$(postgresql_cmd -At -c "select * from $table_name" || :)"
       if [[ "$result" == "$value" ]]; then
         echo "${slave_ip} successfully got value from MASTER ${master_ip}"
         break
       fi
-      if [[ "${i}" == "${max_attempts}" ]]; then
-        echo "The ${slave_ip} failed to see value added on MASTER"
-        echo "Dumping logs for $slave"
-        docker logs $slave
-        return 1
-      fi
+      test $i -lt $max_attempts
       sleep 1
     done
   done
 }
 
+# random_string VARNAME LENGTH=16
+# -------------------------------
+# Set $VARNAME to random string of length $LENGTH.
+random_string ()
+{
+    local val=$(cat /dev/urandom | LC_ALL=C tr -dc "[:alpha:]" | head -c ${2-16})
+    eval $1=\$val
+}
+
+# allocate_docker_network NAME
+# ----------------------------
+# Create docker network usable with --net && --ip parameters later (craeting the
+# network requires explicitly specified --subnet).  Make sure the network of
+# name NAME does not exist, yet.
+allocate_docker_network ()
+{
+  local name pattern found check
+
+  name="$1"
+  pattern=172.18.X.0
+
+  found=false
+  for check in {1..255}
+  do
+    docker network create --driver bridge \
+            --subnet=${pattern//X/$check}/24 "$name" &>/dev/null \
+        || continue
+    found=:
+    DOCKER_NETS="${DOCKER_NETS:-} $name"
+    break
+  done
+
+  # Error when we were unable to find free subnet.
+  $found || return 1
```

Comment:
Right, the '|| return 1' part is also redundant.  Is 'false "can't find free docker network"' enough?